### ISSUE TITLE
ipv6 support

### ISSUE THREAD

**Issue by Robitx** [robinst/linkify#78]
Hey,
I've noticed that IPv6 URLs with bracket notation (specified in https://datatracker.ietf.org/doc/html/rfc2732#section-2 ) are not properly detected.

```
      http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80/index.html
      http://[1080:0:0:0:8:800:200C:417A]/index.html
      http://[3ffe:2a00:100:7031::1]
      http://[1080::8:800:200C:417A]/foo
      http://[::192.9.5.5]/ipng
      http://[::FFFF:129.144.52.38]:80/index.html
      http://[2010:836B:4179::836B:4179]
```

**Comment by robinst** [comment]
Yeah, I wasn't sure if anyone was actually using those. Happy to review a PR to add these if it's not too difficult.

**Comment by andrewsuperlegit** [comment]
@robinst I'm working on it right now- might need to talk to you about an architectural issue because you're currently using ":" to trigger the URL scanner, and its up to you whether I extend the url scanner itself to handle ip6 addresses or if you want me to create an independent ipv6 scanner. dm me if u can get a second

**Comment by robinst** [comment]
I think it should just be part of the URL scanner code that handles hostnames.

**Comment by oooo-ps** [comment]
IPv6 addresses are using at least in CJDNS, Yggdrasil, Mycelium networks, and DNS-less clearnet addresses - please fix it.

**Comment by andrewsuperlegit** [comment]
@oooo-ps my implementation of ipv6 should be merged soon!
